rollup.config.ts

```diff
@@ -35,12 +35,12 @@ const babelPlugin = (type: 'legacy' | 'modern') =>
       type === 'modern'
         ? ''
         : {
-            chrome: 73,
-            firefox: 78,
-            edge: 79,
-            safari: 12,
-            ios: 12,
-            opera: 53,
+            chrome: '73',
+            firefox: '78',
+            edge: '79',
+            safari: '12',
+            ios: '12',
+            opera: '53',
```

Comment:
> > Also the isFetched being false on the first refetch is because createBaseQuery doesnt seem to set the dataUpdateCount property when dehydrating the query on the server. This property is not available on the queryObserver result so we might have to manually set it?
> 
> Hmm, begs the question.. why is the query state that we're hydrating missing some properties? This it the first time I'm actually digging into query-core a bit, and there might be a larger issue?
> 
> I didn't realize that `QueryObserverResult` and `QueryState` are managed separately. Basically, we're hydrating the query and setting the `state`, which is expecting `QueryState`, to a `QueryObserverResult` object in https://github.com/TanStack/query/blob/alpha/packages/solid-query/src/createBaseQuery.ts#L149 (since `observer.subscribe` passes us a `QueryObserverResult`).
> 
> **Should we be subscribing to the query cache instead (since it passes the current query, including state, to the listeners) https://github.com/TanStack/query/blob/alpha/packages/query-core/src/queryObserver.ts#L665?**
> 
> Looking at `QueryState`, 9 of 12 properties overlap with `QueryObserverResult`.
> 
> ```ts
> interface QueryState<TData = unknown, TError = DefaultError> {
>   data: TData | undefined // present on QueryObserverResult
>   dataUpdateCount: number // NOT present on QueryObserverResult
>   dataUpdatedAt: number // present on QueryObserverResult
>   error: TError | null // present on QueryObserverResult
>   errorUpdateCount: number // present on QueryObserverResult
>   errorUpdatedAt: number // present on QueryObserverResult
>   fetchFailureCount: number // present BUT with a different name (failureCount) on QueryObserverResult
>   fetchFailureReason: TError | null // present BUT with a different name (failureReason) on QueryObserverResult
>   fetchMeta: FetchMeta | null // NOT present on QueryObserverResult
>   isInvalidated: boolean // NOT present on QueryObserverResult (perhaps is present but with a different name? not sure what this is)
>   status: QueryStatus // PRESENT on QueryObserverResult
>   fetchStatus: FetchStatus // PRESENT on QueryObserverResult
> }
> ```
> 
> For the immediate issue at hand, the `isFetching` issue is indeed occurring because `dataUpdateCount` is not present on `QueryObserverResult` (interestingly, `errorUpdateCount` IS present on `QueryObserverResult`).
> 
> is the intent to ultimately simplify and consolidate `QueryState` into `QueryObserverResult`? Or does managing `QueryState` as a separate thing serve some other functional purpose? Huge caveat here that I probably don't know what I'm talking about, JUST getting into query-core codebase for the first time 😅.
> 
> For the time being, I'll see if I can get the actual state object hydrating (rather than `QueryObserverResult`).

One reason why they both are not exactly the same, because things like placeholderData or using the `select` function (to transform data) will only affect the data on the observer level. The queryState data is not always equal to the data on the QueryObserverResult.

Reply:
Hmm yeah I can see why query result and query state might each have their own `data` properties. Still not clear why most of the other properties are duplicated between the two, some with slightly different names. A topic for another time though, thanks for merging!